### PR TITLE
[OCPCLOUD-1781]: Ensure CPMS tests work on GCP

### PR DESCRIPTION
**Pull request by odvarkadaniel**
This PR adds unit tests for scaling the machine sizes on GCP cluster. Also checked whether the e2e presubmit test passes - OK, on a working GCP cluster with CPSM in `Active` state.

### CODE REVIEW

File: test/e2e/framework/framework.go [hunk 1]
```diff
@@ -350,6 +354,71 @@ func nextAzureVMSize(current string) (string, error) {
...
+	}
+
+	return nil
+}
+
+// nextGCPVMSize returns the next GCP machine size in the series.
+// The Machine sizes being used are in format <e2|n2|n1>-standard-<number>.
+func nextGCPMachineSize(current string) (string, error) {
+	// Regex to match the GCP machine size string.
+	re := regexp.MustCompile(`(?P<family>[0-9a-z]+)-standard-(?P<version>[0-9]+)`)
```
Comment (JoelSpeed) [inline test/e2e/framework/framework.go:379]: I think this should be multiplier right?
```suggestion
	re := regexp.MustCompile(`(?P<family>[0-9a-z]+)-standard-(?P<multiplier>[0-9]+)`)
```

**Review by JoelSpeed — COMMENTED** [review]
Looks good, couple of nits

File: test/e2e/framework/framework.go [hunk 2]
```diff
@@ -350,6 +354,71 @@ func nextAzureVMSize(current string) (string, error) {
...
+	}
+
+	family := values[1]
+
+	return setNextGCPMachineSize(current, family, multiplier)
+}
+
+// setNextGCPMachineSize returns the new GCP machine size in the series
+// according to the family supported (e2, n1, n2).
+func setNextGCPMachineSize(current, family string, multiplier int) (string, error) {
```
[the remark this answers is not in the document]
Comment (odvarkadaniel) [reply test/e2e/framework/framework.go:401]: Done

File: test/e2e/framework/framework.go (hunk 1 above)
Comment (odvarkadaniel) [reply test/e2e/framework/framework.go]: Makes sense

**Comment by JoelSpeed** [comment]
/approve

I don't think this will pass until #133 is merged

**Comment by JoelSpeed** [comment]
/override ci/prow/e2e-aws-upgrade

This only affects GCP e2e testing, so the AWS failure is unrelated

**Pull request merged**